addons/docs/src/mdx/mdx-compiler-plugin.js

```diff
@@ -52,37 +52,44 @@ function genStoryExport(ast, context) {
   const statements = [];
   const storyKey = getStoryKey(storyName, context.counter);
 
-  let body = ast.children.find(n => n.type !== 'JSXText');
+  const bodyNodes = ast.children.filter(n => n.type !== 'JSXText');
   let storyCode = null;
-
-  if (!body) {
+  let storyVal = null;
+  if (!bodyNodes.length) {
     // plain text node
     const { code } = generate(ast.children[0], {});
     storyCode = `'${code}'`;
+    storyVal = `() => (
+      ${storyCode}
+    )`;
   } else {
-    if (body.type === 'JSXExpressionContainer') {
-      // FIXME: handle fragments
-      body = body.expression;
+    const bodyParts = bodyNodes.map(bodyNode => {
+      const body = bodyNode.type === 'JSXExpressionContainer' ? bodyNode.expression : bodyNode;
+      const { code } = generate(body, {});
+      return { code, body };
+    });
+    // if we have more than two children
+    // 1. Add line breaks
+    // 2. Enclose in <> ... </>
+    storyCode = bodyParts.map(({ code }) => code).join('\n');
+    const storyReactCode = bodyParts.length > 1 ? `<>\n${storyCode}\n</>` : storyCode;
```

Comment:
@shilman - here it is enclosing the elements in `<> </>` if more than one child - is that what you meant for the fragment? 

I think this is more correct to use pure JSX instead of React.Fragment, no?

Reply:
I'll play around with it more. Not what I expected the code to look like, but I also haven't really dug in yet. Thanks for your patience!